src/app/icd/server/BUILD.gn

```diff
@@ -35,7 +42,7 @@ buildconfig_header("icd-server-buildconfig") {
 source_set("icd-server-config") {
   sources = [ "ICDServerConfig.h" ]
 
-  deps = [ ":icd-server-buildconfig" ]
+  public_deps = [ ":icd-server-buildconfig" ]
```

Comment:
I was not able to include the `ICDServerConfig.h` in the `icd-management-server.h` because of the "Can't include this header from here.  " gn error, while for the `ICDManager.h` it was possible. I thought the reason is that this header is not visible in the other directories than `app/icd` and spotted that similar dependencies in other modules uses public_deps. Is there other preferred solution for such a problem?

Reply:
I did a quick test on ToT and i was able to include `ICDServerConfig.h` with `#include <app/icd/server/ICDServerConfig.h>`. Might be changes on the PR blocking this?

Since `ICDServerConfig` is the one that includes `ICDServerBuildConfing, i don't think we should need to modify `deps` to `public_deps`.

Reply:
You are right, it works. I don't know why I observed this issue before, maybe it was only temporary problem due to some other factors.